notes/stv/20220316-test-deploy-01.txt

```diff
@@ -0,0 +1,168 @@
+#   </meta:licence>
+# </meta:header>
+#
+
+
+    Target:
+
+        Run Test Deploy, and run benchmark
+        Validate that the tests fail correctly
+
+    Result:
+  
+        SUCCESS
+
+
+
+
+# -----------------------------------------------------
+# Fetch target branch
+#[user@desktop]
+
+    source "${HOME:?}/aglais.env"
+    pushd "${AGLAIS_CODE}"
+      	git checkout 'feature/enable-arrow'
+
+    popd
+
+
+
+
+# -----------------------------------------------------
+# Create a container to work with.
+#[user@desktop]
+
+    source "${HOME:?}/aglais.env"
+
+    podman run \
+        --rm \
+        --tty \
+        --interactive \
+        --name ansibler \
+        --hostname ansibler \
+        --publish 3000:3000 \
+        --publish 8088:8088 \
+        --env "SSH_AUTH_SOCK=/mnt/ssh_auth_sock" \
+        --volume "${SSH_AUTH_SOCK}:/mnt/ssh_auth_sock:rw,z" \
+        --volume "${HOME:?}/clouds.yaml:/etc/openstack/clouds.yaml:ro,z" \
+        --volume "${AGLAIS_CODE:?}/deployments:/deployments:ro,z" \
+        atolmis/ansible-client:2021.08.25 \
+        bash
+
+
+
+# -----------------------------------------------------
+# Set the cloud and configuration.
+#[root@ansibler]
+
+    cloudname=iris-gaia-red
+
+    configname=zeppelin-27.45-spark-6.27.45
+
+
+
+# -----------------------------------------------------
+# Delete everything.
+#[root@ansibler]
+
+    time \
+        /deployments/openstack/bin/delete-all.sh \
+            "${cloudname:?}"
+
+        > Done
+
+
+
+# -----------------------------------------------------
+# Create everything, using the new config.
+#[root@ansibler]
+
+    time \
+        /deployments/hadoop-yarn/bin/create-all.sh \
+            "${cloudname:?}" \
+            "${configname:?}"   \
+        | tee /tmp/create-all.log
+
+
+        > Done
+
+
+
+# -----------------------------------------------------
+# Run Quick test
+#[root@ansibler]
+
+    num_users=1
+    concurrent=False
+    test_level=quick
+
+    time \
+        /deployments/hadoop-yarn/bin/run-tests.sh \
+            "${cloudname:?}" \
+            "${configname:?}" \
+            "${test_level:?}"  \
+	     ${concurrent:?}  \
+	     ${num_users:?}  \
+        | tee /tmp/run-tests-quick.log
+
+
+
+
+# Results (Formatted): 
+
+------------ Test Result: [PASS] ------------
+
+[{
+	'SetUp': {
+		'totaltime': '44.95',
+		'status': 'SUCCESS',
+		'msg': '',
+		'valid': 'TRUE'
+	},
+	'Mean_proper_motions_over_the_sky': {
+		'totaltime': '52.71',
+		'status': 'SUCCESS',
+		'msg': '',
+		'valid': 'TRUE'
+	},
+	'Source_counts_over_the_sky.json': {
+		'totaltime': '21.34',
+		'status': 'SUCCESS',
+		'msg': '',
+		'valid': 'TRUE'
+	},
+	'Library_Validation.json': {
+		'totaltime': '7.95',
+		'status': 'SUCCESS',
+		'msg': '',
+		'valid': 'TRUE'
+	}
+}]
+
+
+
+# Test that apache arrow is enabled:
+
+# Create a new notebook in Zeppelin:
+
+%pyspark
+spark.conf.get("spark.sql.execution.arrow.pyspark.enabled")
+
+> "true"
```

Comment:
Can we put this in our library check test, and verify that the test fails when this returns `false`.
Just in case this gets changed back by accident.

Reply:
I've added a cell to validate that arrow is enabled to the library validation notebook. (See: https://github.com/wfau/aglais-testing/commit/b501021946ec89963b358f8b6a557d07c0914b1f)

Reply:
:+1: thanks